### PR TITLE
Adds server-compare and browserify, and updates sorting of grouped results accordingly.

### PR DESCRIPTION
**Pull request by jaredgalanis**
The sorting of the table of contents was out of order because of the way the semantic versioning numbering occurs (e.g. a normal sort treats 2.2 > 2.11). 

Using the semver-compare package (and browserify to import) which are used in the api-docs this sorting is now done properly.

The Upcoming Features group is moved to the end of the ordering scheme. 

Partially addresses #15, that issue still needs the grouped deprecations to be sorted.

### CODE REVIEW

**Comment by rwjblue** [comment]
Generally speaking, we should try to avoid browserify. In this case the underlying implementation of semver-compare is _very_ small:

```
function cmp (a, b) {
    var pa = a.split('.');
    var pb = b.split('.');
    for (var i = 0; i < 3; i++) {
        var na = Number(pa[i]);
        var nb = Number(pb[i]);
        if (na > nb) return 1;
        if (nb > na) return -1;
        if (!isNaN(na) && isNaN(nb)) return 1;
        if (isNaN(na) && !isNaN(nb)) return -1;
    }
    return 0;
};
```

Let’s either inline it or use https://github.com/rwjblue/ember-cli-cjs-transform to app.import it...

**Comment by jaredgalanis** [comment]
Thanks @rwjblue! Updating to ember-cli-cjs-transform now.

File: package.json [hunk 1]
```diff
@@ -21,9 +21,11 @@
     "broccoli-asset-rev": "^2.4.5",
     "broccoli-static-site-json": "^1.3.0",
     "ember-ajax": "^3.0.0",
+    "ember-browserify": "^1.2.1",
```
Comment (rwjblue) [inline package.json]: Probably don’t need this any longer

**Comment by mansona** [comment]
An alternative is to just app.import it directly like we do in the Guides App: https://github.com/ember-learn/guides-app/blob/master/package.json#L24 https://github.com/ember-learn/guides-app/blob/master/ember-cli-build.js#L72 where it becomes a global. 

@rwjblue is ember-cli-cjs-transform the **recommended** way to include this stuff? If so then I will probably convert the Guides App over too 👍

File: package.json (hunk 1 above)
Comment (jaredgalanis) [reply package.json]: Ack! Yeah, sorry I missed that. I only ran npm uninstall.

Comment (jaredgalanis) [reply package.json]: Ack! Yeah, sorry I missed that. I just ran npm uninstall.

**Comment by mansona** [comment]
@rwjblue I'm merging this but I would still love an answer re: ember-cli-cjs-transform so I can unify between here and guides-app if necessary 👍

**Comment by rwjblue** [comment]
The transform is only needed because semver-compare is in CJS format. If you app imported something that was already using either UMD or globals, it would work without any special transformation. 

tldr; using app.import is good, and sometimes you need a transformation to deal with inconsistencies in the formatting of the incoming file.

**Comment by mansona** [comment]
@rwjblue so to summarise: if we're using app.import() in the guides and it's working as expected it's not currently recommended that we also use ember-cli-cjs-transform?

**Comment by rwjblue** [comment]
Yes, only use a special transform if you need it. 😃
